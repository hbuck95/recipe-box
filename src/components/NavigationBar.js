import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";

import { library } from "@fortawesome/fontawesome-svg-core";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faUtensils } from "@fortawesome/free-solid-svg-icons";

import 'bootstrap/dist/css/bootstrap.css'

import RecipeForm from './RecipeForm';
import RecipeList from './RecipeList';
import RecipeView from './RecipeView';

import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    NavItem,
} from 'reactstrap';

library.add(faUtensils);

export default class NavigationBar extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isOpen: false
        };
    }

    toggle = () => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }

    render() {
        return (
            <Router>
                <div id="navbar">
                    <Navbar color="dark" light expand="md">
                        <NavbarBrand href="/">Recipe Box <FontAwesomeIcon icon="utensils" /></NavbarBrand>
                        <NavbarToggler onClick={this.toggle} />
                        <Collapse isOpen={this.state.isOpen} navbar>
                            <Nav className="ml-auto" navbar>
                                <NavItem>
                                    <Link id="addrecipe" to="/addrecipe">Add Recipe </Link>
                                </NavItem>
                                <NavItem>
                                    <br/>
                                </NavItem>
                                <NavItem>
                                    <Link to="/viewrecipes">View Recipes</Link>
                                </NavItem>
                            </Nav>
                        </Collapse>
                    </Navbar>

                    <Route exact path="/"/>
                    <Route path="/addrecipe" component={RecipeForm} />
                    <Route path="/viewrecipes" component={RecipeList} />
                    <Route path="/viewrecipe/:id" component={RecipeView}/>
                </div>
            </Router>
        );
    }
}