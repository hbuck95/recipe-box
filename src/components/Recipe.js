import React, { Component } from 'react';
import axios from 'axios';
import { Link } from "react-router-dom";
    
import 'bootstrap/dist/css/bootstrap.css'

export default class Recipe extends Component {
    static defaultProps = {
        image: "https://i.imgur.com/bAmSrZZ.png"
    };

    delete = () => {
        const recipe = {
            _id: this.props._id,
        };
        let userResp = window.confirm("Are you sure you want to delete this recipe?")
        
        if(!userResp)
            return;

        axios.delete("http://localhost:5000/recipe/deleteRecipe",{data:recipe}).then(res => {
            console.log(res);
            window.alert("Recipe deleted!");
            this.props.func()
        }).catch(err => console.log(err));
    };

    loadRecipe = () => {
        console.log(`Clicked ${this.props._id}`);
    }

    render() {
        return (
            
            <div id = {this.props._id}>
                <h3>{this.props.name}</h3>
                <img src={this.props.image} width="128" height="128" alt={this.props.name} onClick={() => this.loadRecipe()}></img>
                <br/><br/>
                {/* <h5>{this.props.desc}</h5> */}
                {/* <p><b>Ingredients: </b>{this.props.ingredients}</p> */}
                <button onClick={this.delete}>Remove</button> <Link to={"/viewrecipe/"+this.props._id}><button>View</button></Link>
                <br/><br/><br/>
            </div>
        )};
}