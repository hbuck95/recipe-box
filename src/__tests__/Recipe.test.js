import React from 'react';
import renderer from 'react-test-renderer';
import NavigationBar from '../components/NavigationBar';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import axios from 'axios';
import { Link, BrowserRouter as Router } from "react-router-dom";

import Recipe from '../components/Recipe';

configure({ adapter: new Adapter() });

it('Recipe uses props', () => {
    const recipeProps = {
        _id: "abc123",
        name: "Pancakes",
        desc: "A tasty recipe for pancakes",
        ingredients: "Egg; Milk; Flour",
        image: "https://i.imgur.com/Xl4kEpP.jpg"
    };

    const wrapper = shallow(<Recipe {...recipeProps} />);
    expect(wrapper.instance().props._id).toEqual(recipeProps._id);
    expect(wrapper.instance().props.name).toEqual(recipeProps.name);
    expect(wrapper.instance().props.desc).toEqual(recipeProps.desc);
    expect(wrapper.instance().props.ingredients).toEqual(recipeProps.ingredients);
    expect(wrapper.instance().props.image).toEqual(recipeProps.image);
    expect(wrapper.instance().props).toEqual(recipeProps);
    expect(wrapper).toMatchSnapshot();
});