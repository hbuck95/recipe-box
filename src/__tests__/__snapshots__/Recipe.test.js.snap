// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe uses props 1`] = `ShallowWrapper {}`;
